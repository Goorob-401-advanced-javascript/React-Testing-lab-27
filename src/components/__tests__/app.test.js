// eslint-disable-line no-unused-vars
import React from 'react';// eslint-disable-line no-unused-vars
import { shallow , mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../counter/counter.js';// eslint-disable-line no-unused-vars

describe('<Counter/>' , ()=>{
  it('exists at the start of our application',()=>{
    let app = shallow(<Counter />);
    expect(app.find('section').exists()).toBeTruthy();  
  });
  it('change the state on click',()=>{
    let app = mount(<Counter />);
    let button = app.find('.up');
    button.simulate('click');
    expect(app.state('count')).toBe(1);

  });
  it('renders correctly',()=> {
    const rendersTree = renderer.create(<Counter />).toJSON();
    expect(rendersTree).toMatchSnapshot();
  });
});