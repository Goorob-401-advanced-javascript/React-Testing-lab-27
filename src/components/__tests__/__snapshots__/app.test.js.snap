// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> renders correctly 1`] = `
<div>
  <section>
     counter : 
    0
  </section>
  <p
    className="count "
  />
  <button
    className="up"
    onClick={[Function]}
    type="button"
  >
    <span>
      +
    </span>
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    <span>
      -
    </span>
  </button>
</div>
`;
